### PR TITLE
Manage filenames

### LINKED ISSUE #156: Manage filenames in preprocessing config
All filenames should be managed in a centralized location. The preprocessing config is the correct place for this, as we know all intermediate file locations once we read the config.

As the database::build method should not be dependent on the preprocessing config, the last part of the preprocessing should be to move all relevant files for the build to standard locations.

### PR DESCRIPTION
**Pull request by Taepper**
Resolves #156 and works further towards #7 / #9

### CODE REVIEW

**Comment by Taepper** [comment]
The linter ran without errors. It was a GH Actions internal bug for the failed run:

```
#17 writing layer sha256:46342c6b979d20e8eeabb5a0f9ec906b4604f1999e7e488c84ac9d5106a83597 60.1s done
#17 ERROR: error writing layer blob: GitHub Actions is temporarily unavailable. Please visit https://www.githubstatus.com/ for the status of our services.

```

**Review by fengelniederhammer — APPROVED** [review]
Nice :+1:

File: src/silo/preprocessing/partition.cpp [hunk 1]
```diff
@@ -93,143 +94,119 @@ std::vector<silo::preprocessing::Chunk> mergePangosToChunks(
...
+   const silo::preprocessing::PartitionChunk& partition_chunk
+) const {
+   return hash<uint32_t>()(partition_chunk.partition) +
+          (hash<uint32_t>()(partition_chunk.chunk) << 3) +
+          (hash<uint32_t>()(partition_chunk.chunk) >> 2);
+}
+
+namespace silo::preprocessing {
+
 void Partitions::save(std::ostream& output_file) const {
```
[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo/preprocessing/partition.cpp:205]: load is coming back tomorrow :)

File: src/silo/preprocessing/preprocessing_config.cpp [hunk 2]
```diff
@@ -63,7 +78,150 @@ PreprocessingConfig::PreprocessingConfig(
...
+std::filesystem::path PreprocessingConfig::getPangoLineageDefinitionFilename() const {
+   return pango_lineage_definition_file;
+}
+
+std::filesystem::path PreprocessingConfig::getReferenceGenomeFilename() const {
+   return reference_genome_file;
+}
+
+std::filesystem::path PreprocessingConfig::getMetadataInputFilename() const {
+   return metadata_file.string();
```
[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo/preprocessing/preprocessing_config.cpp:94]: Weird, I force pushed the changes, but the merge is somehow still the old version?

[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo/preprocessing/preprocessing_config.cpp:94]: I include it in the next PR

**Pull request merged**